src/model/index/index-service.ts

```diff
@@ -0,0 +1,56 @@
+import 'reflect-metadata';
+import {IndexesOptions as SequelizeIndexOptions} from 'sequelize';
+
+const INDEXES_KEY = 'sequelize:indexes';
+
+// https://github.com/sequelize/sequelize/blob/beeff96f/types/lib/query-interface.d.ts#L169
+export interface IndexFieldOptions {
+  name: string;
+  length?: number;
+  order?: 'ASC' | 'DESC';
+  collate?: string;
+}
+
+export type IndexOptions = Pick<SequelizeIndexOptions, Exclude<keyof SequelizeIndexOptions, 'fields'>>;
+
+/**
+ * Returns model indexes from class by restoring this
+ * information from reflect metadata
+ */
+export function getIndexes(target: any): any | undefined {
+  const indexes = Reflect.getMetadata(INDEXES_KEY, target);
+
+  // tslint:disable-next-line:prefer-object-spread
+  return indexes && Object.assign([], indexes);
```

Comment:
Array spreading works a bit differently. It expects the value being spreaded to be an iterable, and it simply iterates over it and pushes each value to the array. `indexes` in my implementation is an object with the Array prototype that has named properties in addition to indexed ones. Named properties are indexes that were assigned a name by the user, the ones stored in the array are not named. Array spreading would drop all named properties.
I am not a big fan of the "prefer-object-spread" rule to be honest. Yes, it's cleaner to use spread operator in most cases, but there are situations when `Object.assign()` is perfectly valid (this right here is a good example) and sometimes can actually be more readable.

Reply:
Ah ok, I didn't realize that `indexes` is an object in the first place. Then I ask myself why are you assigning it to an array anyway? Why not keeping it as an object?

Reply:
Well, it's an object with the Array prototype. I use Array signature for dealing with indexes that don't have names assigned to them (simply push to the array and access by index, I also make use of the `length` property). I guess it's one of the JS idiosyncrasies, the fact that you can use an array both as an array and as a regular object. This is because every array in JS is simply an object with the Array prototype. What I did is almost identical to `Object.setPrototypeOf({ ...indexes }, Array.prototype)`. The difference is that the latter wouldn't set property descriptors correctly and the `length` property would end up in the `Object.keys()` output.
Anyway, if you think this is too confusing, I can store named and unnamed indexes on separate objects. This is certainly different from the way you expect to work with arrays and objects in most other languages.

Reply:
Now I can see your point. But yeah, I think it is a bit too implicit. You could also keep storing the numeric and named fields on the very same object. And just storing the last used numeric id additionally. `const indexMeta = {indexes : {}, currentNumericId: 1}`; But its up to you.

Reply:
@RobinBuschmann pushed an update for this. Does this look good now?